Zend/zend_vm_gen.php

```diff
@@ -2818,6 +2818,13 @@ function gen_vm($def, $skel) {
 				if (isset($dsc['type_spec'])) {
 					$orig_op = $dsc['op'];
 					out($f, "\t\tcase $orig_op:\n");
+					// XXX: Copy the specializations for LONG == LONG and DOUBLE != DOUBLE to work for ===/!== as well.
+					// (Those are currently the only specializations)
+					if ($orig_op === 'ZEND_IS_EQUAL') {
+						out($f, "\t\tcase ZEND_IS_IDENTICAL:\n");
+					} elseif ($orig_op === 'ZEND_IS_NOT_EQUAL') {
+						out($f, "\t\tcase ZEND_IS_NOT_IDENTICAL:\n");
+					}
```

Comment:
This is not the right way.
You may try to add corresponding ZEND_VM_HOT_TYPE_SPEC_HANDLER in zend_vm_def.h instead.
May be zend_vm_def.h and zend_vm_gen.php also should be extended to support identical handlers.

Reply:
> May be zend_vm_def.h and zend_vm_gen.php also should be extended to support identical handlers.

Done. I don't see any other optimizations that can make use of that at the moment